cmake-tests/CMakeLists.txt

```diff
@@ -1,8 +1,9 @@
+set(CMAKE_CXX_COMPILER "${PROJECT_BINARY_DIR}/compiler/bin/clang++")
 
-if(COMMAND CMAKE_FORCE_CXX_COMPILER)
-CMAKE_FORCE_CXX_COMPILER("${PROJECT_BINARY_DIR}/compiler/bin/clang++" MCWAMPCC)
-endif()
 add_executable(cmake-test cmake-test.cpp)
+
+add_dependencies(cmake-test world)
```

Comment:
moving `add_dependencies(cmake-test world)` into `if(TARGET hccrt)` block doesn't work for me since cmake-test binary wants .so and .bc files in right places. I'm reworking the change now